libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/detail/search.hpp

```diff
@@ -0,0 +1,153 @@
+//  Copyright (c) 2020 ETH Zurich
+//  Copyright (c) 2014 Grant Mercer
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+//  This file is based on the following cppreference possible implementation:
+//  https://en.cppreference.com/w/cpp/algorithm/ranges/search
+#pragma once
+
+#include <hpx/config.hpp>
+#include <hpx/algorithms/traits/projected.hpp>
+#include <hpx/functional/invoke.hpp>
+#include <hpx/execution/algorithms/detail/predicates.hpp>
+#include <hpx/parallel/util/detail/algorithm_result.hpp>
+#include <hpx/parallel/util/loop.hpp>
+#include <hpx/parallel/util/partitioner.hpp>
+
+
+#include <cstddef>
+#include <iterator>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace hpx { namespace ranges { namespace detail {
+    /// \cond NOINTERNAL
+    template <typename FwdIter, typename Sent>
+    struct search
+      : public hpx::parallel::v1::detail::algorithm<search<FwdIter, Sent>,
+            FwdIter>
+    {
+        search()
+          : search::algorithm("search")
+        {
+        }
+
+        template <typename ExPolicy, typename FwdIter2, typename Sent2,
+            typename Pred, typename Proj1, typename Proj2>
+        static FwdIter sequential(ExPolicy, FwdIter first, Sent last,
+            FwdIter2 s_first, Sent2 s_last, Pred&& op, Proj1&& proj1,
+            Proj2&& proj2)
+        {
+            for (;; ++first)
+            {
+                FwdIter it1 = first;
+                for (FwdIter2 it2 = s_first;; ++it1, ++it2)
+                {
+                    if (it2 == s_last)
+                        return first;
+                    if (it1 == last)
+                        return it1;
+                    if (!hpx::util::invoke(op, hpx::util::invoke(proj1, *it1),
```

Comment:
Would you mind using `HPX_INVOKE` here instead (and in the other `search`-related files? As in https://github.com/STEllAR-GROUP/hpx/pull/4991.